packages/create-video/src/init-v2.ts

```diff
@@ -0,0 +1,267 @@
+	folderName?: string
+) {
+	if (
+		!(await CreateDirectory.assertFolderEmptyAsync({
+			projectRoot,
+			folderName,
+			overwrite: false,
+		}))
+	) {
+		const message = 'Try using a new directory name, or moving these files.';
+		Log.newLine();
+		Log.info(message);
+		Log.newLine();
+		throw new Error(message);
+	}
+}
+
+const shouldUseYarn = (): boolean => {
+	return Boolean(
+		process.env.npm_execpath?.includes('yarn.js') ||
+			process.env.npm_config_user_agent?.includes('yarn')
+	);
+};
+
+const isGitExecutableAvailable = async () => {
+	try {
+		await execa('git', ['--version']);
+		return true;
+	} catch (e) {
+		if (e.errno === 'ENOENT') {
+			Log.warn('Unable to find `git` command. `git` not in PATH.');
+			return false;
+		}
+	}
+};
+
+const initGitRepoAsync = async (
+	root: string,
+	flags: {silent: boolean; commit: boolean} = {silent: false, commit: true}
+) => {
+	// let's see if we're in a git tree
+	try {
+		await execa('git', ['rev-parse', '--is-inside-work-tree'], {
+			cwd: root,
+		});
+		!flags.silent &&
+			Log.info(
+				'New project is already inside of a git repo, skipping git init.'
+			);
+	} catch (e) {
+		if (e.errno === 'ENOENT') {
+			!flags.silent &&
+				Log.warn('Unable to initialize git repo. `git` not in PATH.');
+			return false;
+		}
+	}
+
+	// not in git tree, so let's init
+	try {
+		await execa('git', ['init'], {cwd: root});
+		!flags.silent && Log.info('Initialized a git repository.');
+
+		if (flags.commit) {
+			await execa('git', ['add', '--all'], {cwd: root, stdio: 'ignore'});
+			await execa('git', ['commit', '-m', 'Create a new Remotion app'], {
+				cwd: root,
+				stdio: 'ignore',
+			});
+			await execa('git', ['branch', '-M', 'main'], {
+				cwd: root,
+				stdio: 'ignore',
+			});
+		}
+		return true;
+	} catch (e) {
+		Log.verbose('git error:', e);
+		// no-op -- this is just a convenience and we don't care if it fails
+		return false;
+	}
+};
+
+const resolveProjectRootAsync = async () => {
+	let projectName = '';
+	try {
+		const {answer} = await prompts({
+			type: 'text',
+			name: 'answer',
+			message: 'What would you like to name your app?',
+			initial: 'my-video',
+			validate: (name) => {
+				const validation = CreateDirectory.validateName(
+					path.basename(path.resolve(name))
+				);
+				if (typeof validation === 'string') {
+					return 'Invalid project name: ' + validation;
+				}
+				return true;
+			},
+		});
+
+		if (typeof answer === 'string') {
+			projectName = answer.trim();
+		}
+	} catch (error) {
+		// Handle the aborted message in a custom way.
+		if (error.code !== 'ABORTED') {
+			throw error;
+		}
+	}
+
+	const projectRoot = path.resolve(projectName);
+	const folderName = path.basename(projectRoot);
+
+	assertValidName(folderName);
+
+	await fs.ensureDir(projectRoot);
+
+	await assertFolderEmptyAsync(projectRoot, folderName);
+
+	return [projectRoot, folderName];
+};
+
+const isNodeVersionGreater = () => {
+	return process.versions.node >= '16.0.0';
```

Comment:
Not necessarily the most stable way to check it... e.g `2.0.0` > `16.0.0` ==> true

Alternative way which doesn't rely on version checking:  `(fs.rmSync ?? fs.rmdirSync)()`

Reply:
okay that sounds awesome. :fire: